### PR TITLE
Don't load plugins that require an LMS context if said context is not available

### PR DESCRIPTION
**Pull request by grant-cleary**
I think there's value in making it more explicit that certain plugins require an LMS context to function. This lessens the likelihood that a misconfiguration will cause the HTML editor to break by trying to load a plugin that calls internal LMS functions, and makes this differentiation more obvious to plugin developers.

### CODE REVIEW

File: components/plugins/contentstyler.js [hunk 1]
```diff
@@ -12,6 +12,10 @@ export class ContentStylerPlugin {
 		return name;
 	}
 
+	get requiresLmsContext() {
+		return false;
```
Comment (grant-cleary) [inline components/plugins/contentstyler.js:16]: @JakeProuse, @jschlumkoski... This is something I'm considering doing that will impact the plugin implementations in your repo as well. I also wasn't 100% sure whether Insert Element and Layouts require an LMS context to work properly, so would appreciate your feedback on that. I know Content Styler does not, and Practices definitely does.

File: components/plugin-registry.js [hunk 2]
```diff
@@ -215,13 +218,16 @@ function addTinyMCEPlugin(plugin, pluginName) {
...
+	const hasLmsContext = context.activeOrgEnrollment;
+
 	if (registryMap.has(editor.id)) {
 		const registries = registryMap.get(editor.id);
 		return {
 			commandRegistry: registries.commandRegistry,
 			eventRegistry: registries.eventRegistry,
 			internalAPIWrapper: isPrivileged ? registries.privilegedInternalAPIWrapper : registries.internalAPIWrapper,
-			uiRegistry: registries.uiRegistry
+			uiRegistry: hasLmsContext ? registries.lmsAdaptedUiRegister : registries.uiRegistry
```
Comment (grant-cleary) [inline components/plugin-registry.js:230]: I'm not 100% sure about this differentiation. On the one hand, I think it makes sense to not provide tools plugins can use to shoot themselves in the foot if they accidentally misconfigure their plugin. On the other hand... It's JavaScript and this helpers object is kind of opaque anyways, so does this actually make a difference? 🤷

Certainly it's true that these LMS adapters will all explode if they're used without a valid LMS context, so I still think not passing them in makes more sense.

File: components/plugins/contentstyler.js (hunk 1 above)
Comment (JakeProuse) [reply components/plugins/contentstyler.js:16]: I think our IE and Layouts plugins do need the context for opening a dialog (I seem to remember it being required, may be wrong here) @jschlumkoski ?

**Comment by grant-cleary** [comment]
After chatting with @dbatiste offline, I _think_ this is probably the best solution for now. I don't think it's ideal for the editor to have to know things about the LMS, and ideally these are things that individual plugins would handle themselves... But the existence of FRAs forces the editor to have at least some LMS knowledge, and this overall change at least keeps that centralized.

I would expect this API to change in the future once FRAs are dead, but since we'll be dealing with external plugins by then, I expect we'll have to figure out a bit of a migration path to get there.

File: components/plugins/contentstyler.js (hunk 1 above)
Comment (JakeProuse) [reply components/plugins/contentstyler.js:16]: In that case Layouts may be fine, Insert Element [uses the valence controller](https://github.com/BrightspaceUI/htmleditor/blob/main/components/plugins/insert-element.js#L36) so I think it'll still need it

Comment (JakeProuse) [reply components/plugins/contentstyler.js:16]: Layouts also fires events though, while it doesn't explicitly get cookies they are needed for the request

Comment (grant-cleary) [reply components/plugins/contentstyler.js:16]: Ah okay. Might be that it's safer to classify both of these plugins as requiring the LMS then. It mostly comes down to whether you ever anticipate (or want) your plugins to run in contexts with no backing LMS. There aren't many cases where that's currently a thing, but they exist.

**Review by JakeProuse — APPROVED** [review]

**Pull request merged**